v7/flag_overrides.go

```diff
@@ -0,0 +1,156 @@
+type OverrideBehaviour int
+
+const (
+	// LocalOnly means that when evaluating values, the SDK will not use feature flags and settings from the
+	// ConfigCat CDN, but it will use all feature flags and settings that are loaded from local-override sources.
+	LocalOnly = 0
+
+	// LocalOverRemote means that when evaluating values, the SDK will use all feature flags and settings that are
+	// downloaded from the ConfigCat CDN, plus all feature flags and settings that are loaded from
+	// local-override sources. If a feature flag or a setting is defined both in the fetched and the local-override
+	// source then the local-override version will take precedence.
+	LocalOverRemote = 1
+
+	// RemoteOverLocal means when evaluating values, the SDK will use all feature flags and settings that are
+	// downloaded from the ConfigCat CDN, plus all feature flags and settings that are loaded from local-override
+	// sources. If a feature flag or a setting is defined both in the fetched and the local-override source then the
+	// fetched version will take precedence.
+	RemoteOverLocal = 2
+)
+
+// FlagOverrides describes feature flag and setting overrides. With flag overrides you can overwrite the
+// feature flags downloaded from the ConfigCat CDN with local values.
+//
+// With Values, you can set up the SDK to load your feature flag overrides from a map.
+//
+// With FilePath, you can set up the SDK to load your feature flag overrides from a JSON file.
+type FlagOverrides struct {
+	// Behaviour describes how the overrides should behave. Default is LocalOnly.
+	Behaviour OverrideBehaviour
+
+	// Values is a map that contains the overrides.
+	// Each value must be one of the following types: bool, int, float64, or string.
+	Values map[string]interface{}
+
+	// FilePath is the path to a JSON file that contains the overrides.
+	FilePath string
+
+	entries           map[string]*wireconfig.Entry
+	localOnlySnapshot *Snapshot
+}
+
+func (f *FlagOverrides) preLoad(logger *leveledLogger) {
+	if f.Behaviour != LocalOnly && f.Behaviour != LocalOverRemote && f.Behaviour != RemoteOverLocal {
+		logger.Errorf("flag overrides behaviour configuration is invalid. 'Behavior' is %v.", f.Behaviour)
+		return
+	}
+	if f.Values == nil && f.FilePath == "" {
+		logger.Errorf("flag overrides configuration is invalid. 'Values' or 'FilePath' must be set.")
+		return
+	}
+
+	f.entries = f.loadEntries(logger)
+	f.fixEntries()
+	if f.Behaviour == LocalOnly {
+		f.localOnlySnapshot = f.createLocalOnlySnapshot(logger)
+	}
+}
+
+func (f *FlagOverrides) loadEntries(logger *leveledLogger) map[string]*wireconfig.Entry {
+	if f.Values != nil {
+		entries := make(map[string]*wireconfig.Entry, len(f.Values))
+		for key, value := range f.Values {
+			switch value.(type) {
+			case bool, int, float64, string:
+			default:
+				logger.Errorf("value for flag %q has unexpected type %T (%#v); must be bool, int, float64 or string", key, value, value)
```

Comment:
This causes the entire override to be ignored if it contains one invalid entry, which is arguably overkill.